### PR TITLE
Remove console noise from unit tests.

### PR DESCRIPTION
**Pull request by mykmartin**

### CODE REVIEW

File: src/runtime/testing/tests/sequence-test.ts [hunk 1]
```diff
@@ -57,7 +58,7 @@ describe('Sequence testing infrastructure', async () => {
     flowTest.setEndInvariant(total, (value) => assert.strictEqual(value, 3));
 
     try {
-      await flowTest.test();
+      await ConCap.silence(() => flowTest.test());
```
Comment (Cypher1) [inline src/runtime/testing/tests/sequence-test.ts:61]: ConCap was such a good idea. Thanks Myk

File: src/runtime/tests/manifest-test.ts [hunk 2]
```diff
@@ -3709,10 +3709,10 @@ Only type variables may have '*' fields.
     });
 
     it('warns about using multiple `*` in a single variable constraint', async () => {
-      const manifest = await parseManifest(`
+      const manifest = await ConCap.silence(() => parseManifest(`
```
Comment (Cypher1) [inline src/runtime/tests/manifest-test.ts:3712]: Should we check that the log is being printed or is it not important?

File: src/testing/test-util.ts [hunk 3]
```diff
@@ -86,13 +89,27 @@ export class ConCap {
   }
 
   /** Discards all calls to console.log and its friends. Returns the result of fn. */
-  static silence<T>(fn: () => T): T {
+  // tslint:disable-next-line: no-any
+  static silence<T>(fn: () => any): any {
```
Comment (Cypher1) [inline src/testing/test-util.ts:93]: Probably doesn't need `<T>` if we're using any now.

File: src/types/tests/refiner-test.ts [hunk 4]
```diff
@@ -492,7 +492,6 @@ describe('refiner', () => {
             particle Foo
                 input: reads Something {num: Number} [ (2+11-9) > num or False ]
         `);
-        viewAst(manifestAst1);
```
Comment (Cypher1) [inline src/types/tests/refiner-test.ts:495]: Sorry! Missed this in my usual grep console to make sure I don't add logs. Thanks for catching it

**Review by Cypher1 — APPROVED** [review]
LGTM (one actual nit, non-blocking but would be nice)

File: src/testing/test-util.ts [hunk 5]
```diff
@@ -70,12 +70,15 @@ export class ConCap {
 
     const result = fn();
     if (result && typeof result.then === 'function') {
-      let resolve;
-      const wrap = new Promise(r => resolve = r);
+      let resolve, reject;
```
Comment (Cypher1) [inline src/testing/test-util.ts:73]: Linter needs: 
```
let resolve;
let reject;
```

File: src/testing/test-util.ts [hunk 6]
```diff
@@ -70,12 +70,15 @@ export class ConCap {
...
+      const wrap = new Promise((res, rej) => [resolve, reject] = [res, rej]);
       result.then(value => {
         cc.result = value;
         cc.restore();
         resolve(cc);
+      }, err => {
+        cc.restore();
+        reject(err)
```
Comment (Cypher1) [inline src/testing/test-util.ts:81]:
```
; // Missing semi
```

File: src/testing/test-util.ts [hunk 7]
```diff
@@ -86,13 +89,27 @@ export class ConCap {
...
-    return result;
+    if (result && typeof (result as any).then === 'function') {
+      let resolve, reject;
+      const wrap = new Promise((res, rej) => [resolve, reject] = [res, rej]);
+      result.then(value => {
+        cc.restore();
+        resolve(value);
+      }, err => {
+        cc.restore();
+        reject(err)
```
Comment (Cypher1) [inline src/testing/test-util.ts:106]: Same here (missing semi)

File: src/testing/test-util.ts [hunk 8]
```diff
@@ -86,13 +89,27 @@ export class ConCap {
...
+  // tslint:disable-next-line: no-any
+  static silence(fn: () => any): any {
     const cc = new ConCap();
     console.log = console.warn = console.error = console.dir = () => {};
 
     const result = fn();
-    cc.restore();
-    return result;
+    if (result && typeof (result as any).then === 'function') {
+      let resolve, reject;
```
Comment (Cypher1) [inline src/testing/test-util.ts:99]: Same here, multi-let

**Comment by mykmartin** [comment]
Hey Josh, thanks for the lint fixes but I ended up restructuring the ConCap class a bit anyway (got tired of fixing the same thing in two places, several times...)

**Comment by Cypher1** [comment]
No problem, just made the suggestions and then saw the merge button and though 'hey, may as well get the build going'. Thanks Myk

**Pull request merged**